eZ/Publish/Core/Persistence/Legacy/URL/Query/CriterionHandler/Base.php

```diff
@@ -8,59 +8,82 @@
 
 namespace eZ\Publish\Core\Persistence\Legacy\URL\Query\CriterionHandler;
 
-use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use Doctrine\DBAL\Query\QueryBuilder;
+use eZ\Publish\Core\Persistence\Legacy\Content\Gateway as ContentGateway;
+use eZ\Publish\Core\Persistence\Legacy\URL\Gateway\DoctrineDatabase;
 use eZ\Publish\Core\Persistence\Legacy\URL\Query\CriterionHandler;
 
 abstract class Base implements CriterionHandler
 {
     /**
      * Inner join `ezurl_object_link` table if not joined yet.
-     *
-     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
      */
-    protected function joinContentObjectLink(SelectQuery $query): void
+    protected function joinContentObjectLink(QueryBuilder $query): void
     {
-        if (strpos($query->getQuery(), 'INNER JOIN ezurl_object_link ') === false) {
+        if (false === $this->hasJoinedTable($query, DoctrineDatabase::URL_LINK_TABLE)) {
```

Comment:
```suggestion
        if (!$this->hasJoinedTable($query, DoctrineDatabase::URL_LINK_TABLE)) {
```

Reply:
Hmm, the point here was readability, but I don't mind really. Fixed via ec0ac44.